azure/datalake/store/lib.py

```diff
@@ -282,35 +282,35 @@ def call(self, op, path='', **kwargs):
         func = getattr(requests, method)
         url = self.url + path
         try:
-            headers = self.head
+            headers = self.head.copy()
```

Comment:
Thanks for catching this @jbcrail! Is there any where else where we should take a look to make sure we aren't modifying static members during concurrent calls?